contracts/Vault.vy

```diff
@@ -855,11 +867,15 @@ def deposit(_amount: uint256 = MAX_UINT256, recipient: address = msg.sender) ->
 @internal
 def _shareValue(shares: uint256) -> uint256:
     # Determines the current value of `shares`.
-    # NOTE: if sqrt(Vault.totalAssets()) > 1e37, this could potentially revert
-    # NOTE: using 1e3 for extra precision here, when decimals is low
-    return ((10 ** 3 * (shares * self._totalAssets())) / self.totalSupply) / 10 ** 3
-
-
+        # NOTE: if sqrt(Vault.totalAssets()) >>> 1e39, this could potentially revert
+    lockedFundsRatio: uint256 = (block.timestamp - self.lastReport) * self.lockedProfitDegration
+    freeFunds: uint256 = self._totalAssets() 
+    if(lockedFundsRatio < convert(1e18, uint256)):
+        freeFunds = freeFunds - (self.lockedProfit - (lockedFundsRatio * self.lockedProfit / convert(1e18, uint256)))
```

Comment:
Can probably turn this into a `min(freeFunds, ...)` statement

Reply:
This at least makes things a little cleaner (the exact same code)
```suggestion
        freeFunds -= self.lockedProfit - (lockedFundsRatio * self.lockedProfit / convert(1e18, uint256))
```

Reply:
Also, why bother doing `convert(1e18, uint256)` when you can just do `10 ** 18`?

Reply:
I think the 10 ** 18 should be a constant define on top

Reply:
resolved in latest